assets/src/BlockEditorValidation.js

```diff
@@ -0,0 +1,93 @@
+const { registerPlugin } = wp.plugins;
+const { PluginPrePublishPanel } = wp.editPost;
+const { dispatch } = wp.data;
+const {__} = wp.i18n;
+
+export const blockEditorValidation = () => {
+  registerPlugin( 'pre-publish-checklist', { render: PrePublishCheckList } );
+};
+
+const PrePublishCheckList = () => {
+  let lockPost = false;
+
+  let checkListMsg = '';
+  let classname    = '';
+
+  // "Analytics & Tracking" fields validation.
+  let $htmlbody   = $( 'html, body' );
+  // Filter CMB2 fields that have the 'data-validation' attribute set to 'required'.
+  let $toValidate = $( '[data-validation]' );
+
+  let analyticsErrorFlag   = false;
+
+  if ( $toValidate.length ) {
+    let $row = null;
+
+    $toValidate.each(function () {
+      const $this = $(this);
+      const val = $this.val();
+      $row = $this.parents('.cmb-row');
+
+      if ($this.is('[type="button"]') || $this.is('.cmb2-upload-file-id')) {
+        return true;
+      }
+
+      // Apply validation only for campaign post types.
+      if ('campaign' === $('#post_type').val() && 'required' === $this.data('validation')) {
+        if ($row.is('.cmb-type-file-list')) {
+
+          const has_LIs = $row.find('ul.cmb-attach-list li').length > 0;
+
+          if (!has_LIs) {
+            $row.find('select').addClass('cmb2_required_field_error');
+            analyticsErrorFlag = true;
+          } else {
+            $row.find('select').removeClass('cmb2_required_field_error');
+          }
+
+        } else {
+          if (!val || 'not set' === val) {
+            $row.find('select').addClass('cmb2_required_field_error');
+            analyticsErrorFlag = true;
+          } else {
+            $row.find('select').removeClass('cmb2_required_field_error');
+          }
+        }
+      }
+    });
+
+    if ( analyticsErrorFlag ) {
+      lockPost = analyticsErrorFlag;
+      // Open "Analytics & Tracking" fields postbox, if closed.
+      const p4_postbox = $htmlbody.find('#p4_campaign_fields');
+      if (p4_postbox.hasClass('closed')) {
+        p4_postbox.removeClass('closed');
+      }
+      checkListMsg += __( 'Please check "Analytics & Tracking" section for required fields.', 'planet4-master-theme-backend' );
```

Comment:
Ah hadn't noticed it so this must be why it now has the specific `analyticsErrorFlag` variable. But maybe here it would make more sense to list all invalid fields instead of only the section? 

So that editors instantly see all that is wrong with the post, without having to navigate to the analytics section.

Then we wouldn't need to have a variable for each section and it will be easier to add other fields.

Reply:
Done.
![image](https://user-images.githubusercontent.com/5357471/80509339-e8ad6980-8996-11ea-9418-f17bed659c73.png)
Please have a look.